spec/features/gardens/edit_spec.rb

```diff
@@ -45,18 +45,29 @@
                           email: '123@gmail.com' },
                       relationships: {
                           gardens: {
-                              data: [ @garden ] }}})
-
-      @garden = @user.gardens.first
+                              data: [ {id: "4", type: "garden"} ] }}})
 
       garden1 = File.read('spec/fixtures/public_garden.json')
-      stub_request(:get, "https://solar-garden-be.herokuapp.com/api/v1/gardens/3").to_return(status: 200, body: garden1)
+      stub_request(:get, "https://solar-garden-be.herokuapp.com/api/v1/gardens/#{@garden[:id]}").to_return(status: 200, body: garden1)
 
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
     end
 
-    it 'can visit edit a garden page' do
-      visit "/gardens/#{@garden[:id]}/edit"
+    it 'can visit edit a garden page from dashboard' do
+      visit dashboard_path
+
+      # it seems like this syntax isn't actually clicking the edit icon
+      find('.fa-edit').click
```

Comment:
can't get this and test below to pass. It seems like the element isn't actually being clicked w/ this syntax

Reply:
For the garden show page, we could do something like this:

```ruby
within('.garden-info') do 
  first(:link).click
end
```

For the dashboard since edit isn't the first link, trying to find a workaround.